custodial-copy-backend/src/main/scala/uk/gov/nationalarchives/custodialcopy/Processor.scala

```diff
@@ -236,10 +236,12 @@ class Processor(
     case fo: FileObject =>
       for {
         writePath <- fo.sourceFilePath(config.workDir)
```

Comment:
Should we write the path after we determine whether the we are going to download the file and write to it, (I was thinking about the speed of IO operations) or should does it not matter?

Reply:
It's a good point. It won't work as so I've changed the case class to make the source path `Option` and then just check if the option is defined which is a bit cleaner.